src/main/java/frc/robot/subsystems/Indexer.java

```diff
@@ -0,0 +1,220 @@
+ * Shooter in that it should only put the balls in when the Shooter is ready The
+ * indexer is triggered with motors and uses belts to move the balls(controller)
+ * The indexer will use IR sensors so we can see the presence of a ball The
+ * distance that the balls are pushed between the belts needs to be coded so
+ * that they can move fluidly When a ball is at the top of the Pierre, the ball
+ * at the entrance of Pierre needs to be moved forward at the same pace of the
+ * balls being intaked
+ * 
+ */
+
+public class Indexer extends SubsystemBase {
+    final SendableSpeedController singulator;
+    private final SendableSpeedController pierreMotor;
+    final BooleanSupplier frontIntakeIR;
+    final BooleanSupplier bottomPierreIR;
+    final BooleanSupplier topPierreIR;
+    final BooleanSupplier backIntakeIR;
+    public double ballCounting;
+
+    private static final double singulatorMotorSpeed = 0.95;
+    private static final double pierreIndexSpeed = 0.85;
+
+    public Indexer(final IndexMap map) {
+        super();
+        singulator = map.singulator();
+        frontIntakeIR = map.frontIntakeIR();
+        bottomPierreIR = map.bottomPierreIR();
+        topPierreIR = map.topPierreIR();
+        backIntakeIR = map.backIntakeIR();
+        pierreMotor = map.pierreMotor();
+
+    }
+
+    public SequentialCommandGroup intakeToPierre() {
+        return new SequentialCommandGroup(pierrePossesion(), runToClearBottomSensor());
+    }
+
+    public SequentialCommandGroup shootOneBall() {
+        return new SequentialCommandGroup(loadBallToTop(), unLoadBall());
+    }
+
+    public ParallelDeadlineGroup topSensorTriggered() {
+        final ParallelDeadlineGroup topIRStop = new ParallelDeadlineGroup(
+                new SequentialCommandGroup(pierrePossesion(), runToClearBottomSensor()));
+        topIRStop.setDeadline(stopWhenBallsAtTop());
+        return topIRStop;
+    }
+    // This will stop all commands when the top IR sensor on Pierre is triggered
+
+    public CommandBase indexMotor(final double motorSpeed) {
+        return new StartEndCommand(() -> {
+            singulator.set(motorSpeed);
+        }, () -> {
+            singulator.set(0);
+        }, this);
+    }
+
+    public CommandBase quicklyPush() {
+        return indexMotor(singulatorMotorSpeed);
+    }
+
+    public CommandBase reversePush() {
+        return indexMotor(-singulatorMotorSpeed);
+    }
+
+    public CommandBase quicklyOutput() {
+        return indexMotor(pierreIndexSpeed);
+    }
+
+    public CommandBase reverseOutput() {
+        return indexMotor(-pierreIndexSpeed);
+    }
+
+    public CommandBase singulatorPossesion() {
+        return new FunctionalCommand(() -> {
+
+        }, () -> {
+
+        }, (interrupted) -> {
+
+            singulator.set(0);
+
+        }, () -> {
+
+            return frontIntakeIR.getAsBoolean() || topPierreIR.getAsBoolean();
+
+        }, this);
```

Comment:
Try using `WaitUntilCommand`, which takes just the `isFinished` lambda, and then adding `andThen` afterwards to clear the singulator.